src/element.candlestick.js

```diff
@@ -2,139 +2,46 @@
 
 module.exports = function(Chart) {
 
-	var helpers = Chart.helpers,
-		globalOpts = Chart.defaults.global,
-		defaultColor = globalOpts.defaultColor;
-
-	globalOpts.elements.candlestick = {
-		upCandleColor: "rgba(80, 160, 115, 1)",
-		downCandleColor: "rgba(215, 85, 65, 1)",
-		outlineCandleColor: "rgba(90, 90, 90, 1)",
-		outlineCandleWidth: 1,
-	};
-
-	function isVertical(bar) {
-		return bar._view.width !== undefined;
-	}
-
-	/**
-	 * Helper function to get the bounds of the candle
-	 * @private
-	 * @param bar {Chart.Element.Candlestick} the bar
-	 * @return {Bounds} bounds of the bar
-	 */
-	function getBarBounds(candle) {
-		var vm = candle._view;
-		var x1, x2, y1, y2;
-
-		var halfWidth = vm.width / 2;
-		x1 = vm.x - halfWidth;
-		x2 = vm.x + halfWidth;
-		y1 = vm.candle.h;
-		y2 = vm.candle.l;
-
-
-		return {
-			left: x1,
-			top: y1,
-			right: x2,
-			bottom: y2
-		};
-	}
+	var helpers = Chart.helpers;
+	var globalOpts = Chart.defaults.global;
+
+	globalOpts.elements.candlestick = Object.assign(globalOpts.elements.financial, {
+		color: Object.assign(globalOpts.elements.financial.color, {
+			outline: globalOpts.elements.financial.color.linear,
+		}),
+		outlineWidth: 1,
+	});
 
-	Chart.elements.Candlestick = Chart.Element.extend({
+	Chart.elements.candlestick = Chart.elements.financial.extend({
 		draw: function() {
 			var ctx = this._chart.ctx;
 			var vm = this._view;
-			var left, right, top, bottom, signX, signY, borderSkipped;
-			var borderWidth = vm.borderWidth;
-
 
 			var x = vm.x;
 			var o = vm.candle.o;
 			var h = vm.candle.h;
 			var l = vm.candle.l;
 			var c = vm.candle.c;
 
-			ctx.strokeStyle = helpers.getValueOrDefault(vm.outlineCandleColor, globalOpts.elements.candlestick.outlineCandleColor);
-			ctx.lineWidth = helpers.getValueOrDefault(vm.outlineCandleWidth, globalOpts.elements.candlestick.outlineCandleWidth);
+			ctx.strokeStyle = helpers.getValueOrDefault(vm.color?vm.color.outline:undefined, globalOpts.elements.candlestick.color.outline);
```

Comment:
we should add a space on either side of the `?` and `:`.  we were looking at enforcing the chart.js style rules on all the plugin repos, so I'm hoping to stay as close to their rules as possible